### ISSUE TITLE
Test Explorer shows excess class name for tests with multiple TestCaseAttribute

### ISSUE THREAD

**Issue by Deni35** [nunit/nunit#3340]
Look at `UnitTests1.Test_3`. What is that class name for? Is it bug or feature?

![Test Explorer](https://user-images.githubusercontent.com/7755015/62620960-4d192c00-b923-11e9-82c2-87f52e10ed29.png)

```
public class UnitTests1 {

    [TestCase( 0 )]
    public void Test_1(int id) {
    }

    [Test]
    public void Test_2() {
    }

    [TestCase( 666 )]
    [TestCase( 777 )]
    public void Test_3(int id) {
    }
    //[TestCase( null )]
    //public void Test_3(string id) {
    //}

    [Test]
    public void Test_4() {
    }

    [Test]
    public void Test_5() {
    }

}
```

**Comment by CharliePoole** [comment]
It's clearly a bug. In our old GUI, for example, we would show that as just "Test3". The TestCentric GUI shows it the same way. As explained in #3341, I believe this is a TestExplorer problem. @OsirisTerje Could you review?

**Comment by OsirisTerje** [comment]
@CharliePoole I'll check.

**Comment by Deni35** [comment]
The same problem when I have few overloads:

    [TestCase( 777 )]
    public void Test_3(int id) {
    }
    [TestCase( null )]
    public void Test_3(string id) {
    }

**Comment by mikkelbu** [comment]
@Deni35 Which version of Visual Studio are you using? I have VS 16.2.0 Preview 1.0 and it looks like this
<img src="https://user-images.githubusercontent.com/1007631/62652096-d3049980-b95a-11e9-8038-971854db717f.png" width="50%" height="50%">

**Comment by Deni35** [comment]
> @Deni35 Which version of Visual Studio are you using? I have VS 16.2.0 Preview 1.0 and it looks like this
> <img alt="" width="50%" height="50%" src="https://user-images.githubusercontent.com/1007631/62652096-d3049980-b95a-11e9-8038-971854db717f.png">

16.3.3 Preview 1.0

**Comment by mikkelbu** [comment]
After an update to 16.3.0 preview 1.0 I get the same problem, so this issue is probably due to an updated vstest in Visual Studio.
<img src="https://user-images.githubusercontent.com/1007631/62653750-858a2b80-b95e-11e9-9108-3f66b81db3e3.png" width="50%" height="50%">

**Comment by mikkelbu** [comment]
I'm creating the issue now

**Comment by mikkelbu** [comment]
@OsirisTerje Should we move this to the adapter project (even though it is a problem in VS)?

**Comment by OsirisTerje** [comment]
@mikkelbu  Yes, please do.  We tag it with VSIssue, and we can follow up with the PGs.
